src/shared_modules/utils/tests/rocksDBIterator_test.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Wazuh - Shared Modules utils tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 9, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "rocksDBIterator_test.hpp"
+
+TEST_F(RocksDBIteratorTest, TestBasicIteration)
+{
+    // Insert some elements to the database.
+    rocksDb->Put(rocksdb::WriteOptions(), "key1", "value1");
+    rocksDb->Put(rocksdb::WriteOptions(), "key2", "value2");
+    rocksDb->Put(rocksdb::WriteOptions(), "key3", "value3");
+
+    auto db_iterator {Utils::RocksDBIterator(rocksDb)};
```

Comment:
Here, we get the iterator from `std::unique_ptr<rocksdb::DB>`.
But how would we get the iterator from a  `Utils::RocksDBWrapper` object?

We don't have a getter method to obtain the `m_db` instance, and the class `RocksDBWrapper` doesn't have a method to retrieve an iterator either.